### ISSUE TITLE
CereVoice 2-5s TTS delays

### ISSUE THREAD

**Issue by Catreina** [EDCD/EDDI#2030]
## EDDI version in which issue is found
Current

## VoiceAttack version in which issue is found (as applicable)
1.8.7 (but it happens when not using as a plugin too)

## Steps to reproduce
Install VA. Install EDDI. Install CereVoice Caitlin/Heather (I do not have any other voices to test with). Attempt to use voice with EDDI.

## Expected
Near instant synthesis, as with Windows default voices

## Observed
2-5s delays consistently

## Investigation
Fresh installs of all three software. Testing in windows TTS (no issues). 

Snippet from log where you can see a 2s delay with CereVoice Caitlin between setting Speak and adding effects (and no delay with Eva):

2020-12-03T18:02:28 [Debug] SpeechService:speak Selecting voice CereVoice Caitlin - English (Ireland)
2020-12-03T18:02:28 [Debug] SpeechService:speak Feeding normal text to synthesizer: This is how I will sound in your Adder.
2020-12-03T18:02:28 [Debug] SpeechService:speak {"standardVoice":"CereVoice Caitlin - English (Ireland)","volume":65,"effectsLevel":25,"distortOnDamage":true,"rate":0,"disablessml":true,"enableicao":true}
2020-12-03T18:02:30 [Debug] SpeechFX:addEffectsToSource Extending duration by 500ms
2020-12-03T18:02:30 [Debug] SpeechService:Speak Stream length is 123334
2020-12-03T18:02:30 [Debug] SpeechFX:addEffectsToSource Effects level is 25, chorus level is 15, reverb level is 20, echo delay is 50
2020-12-03T18:02:30 [Debug] SpeechService:Speak Seeking back to the beginning of the stream
2020-12-03T18:02:30 [Debug] SpeechService:StartSpeech We can - setting active speech
2020-12-03T18:02:30 [Debug] SpeechService:StartSpeech Checking to see if we can start speech
2020-12-03T18:02:30 [Debug] SpeechService:play Starting speech
2020-12-03T18:02:30 [Debug] SpeechService:StartSpeech Playing sound buffer
2020-12-03T18:02:30 [Debug] SpeechService:play Waiting for speech - 00:00:03.2960000
2020-12-03T18:02:33 [Debug] SpeechService:play Finished waiting for speech
2020-12-03T18:02:33 [Debug] SpeechService:StopCurrentSpeech Disposing of active speech
2020-12-03T18:02:33 [Debug] SpeechService:StopCurrentSpeech Stopping active speech
2020-12-03T18:02:33 [Debug] SpeechService:StopCurrentSpeech Stopped current speech
2020-12-03T18:02:35 [Debug] SpeechService:Say Interrupting current speech
2020-12-03T18:02:35 [Debug] SpeechService:speak Selecting voice Microsoft Eva Mobile
2020-12-03T18:02:35 [Debug] SpeechService:speak Feeding normal text to synthesizer: This is how I will sound in your Adder.
2020-12-03T18:02:35 [Debug] SpeechService:speak {"standardVoice":"Microsoft Eva Mobile","volume":65,"effectsLevel":25,"distortOnDamage":true,"rate":0,"disablessml":true,"enableicao":true}
2020-12-03T18:02:35 [Debug] SpeechFX:addEffectsToSource Effects level is 25, chorus level is 15, reverb level is 20, echo delay is 50
2020-12-03T18:02:35 [Debug] SpeechService:Speak Seeking back to the beginning of the stream
2020-12-03T18:02:35 [Debug] SpeechFX:addEffectsToSource Extending duration by 500ms
2020-12-03T18:02:35 [Debug] SpeechService:Speak Stream length is 95592
2020-12-03T18:02:36 [Debug] SpeechService:play Starting speech
2020-12-03T18:02:36 [Debug] SpeechService:StartSpeech Checking to see if we can start speech
2020-12-03T18:02:36 [Debug] SpeechService:StartSpeech Playing sound buffer
2020-12-03T18:02:36 [Debug] SpeechService:StartSpeech We can - setting active speech
2020-12-03T18:02:36 [Debug] SpeechService:play Waiting for speech - 00:00:02.6670000
2020-12-03T18:02:38 [Debug] SpeechService:StopCurrentSpeech Stopping active speech
2020-12-03T18:02:38 [Debug] SpeechService:StopCurrentSpeech Disposing of active speech
2020-12-03T18:02:38 [Debug] SpeechService:play Finished waiting for speech
2020-12-03T18:02:38 [Debug] SpeechService:StopCurrentSpeech Stopped current speech

**Comment by Tkael** [comment]
I don't have Cerevoice Caitlin, but here are my results with Ivona Amy and Cerevoice Heather in standalone mode (EDDI only):
```
2020-12-04T14:07:34 [Debug] SpeechService:Say Interrupting current speech
2020-12-04T14:07:34 [Debug] SpeechService:speak Selecting voice IVONA 2 Amy OEM
2020-12-04T14:07:34 [Debug] SpeechService:speak {"standardVoice":"IVONA 2 Amy OEM","volume":80,"effectsLevel":35,"distortOnDamage":false,"rate":0,"disablessml":false,"enableicao":false}
2020-12-04T14:07:34 [Debug] SpeechService:speak Feeding normal text to synthesizer: This is how I will sound in your Adder.
2020-12-04T14:07:34 [Debug] SpeechService:Speak Seeking back to the beginning of the stream
2020-12-04T14:07:34 [Debug] SpeechService:Speak Stream length is 96286
2020-12-04T14:07:34 [Debug] SpeechFX:addEffectsToSource Effects level is 35, chorus level is 21, reverb level is 28, echo delay is 50
2020-12-04T14:07:34 [Debug] SpeechFX:addEffectsToSource Extending duration by 500ms
2020-12-04T14:07:34 [Debug] SpeechService:play Starting speech
2020-12-04T14:07:34 [Debug] SpeechService:StartSpeech Playing sound buffer
2020-12-04T14:07:34 [Debug] SpeechService:StartSpeech Checking to see if we can start speech
2020-12-04T14:07:34 [Debug] SpeechService:StartSpeech We can - setting active speech
2020-12-04T14:07:34 [Debug] SpeechService:play Waiting for speech - 00:00:02.6820000
2020-12-04T14:07:37 [Debug] SpeechService:play Finished waiting for speech
2020-12-04T14:07:37 [Debug] SpeechService:StopCurrentSpeech Stopping active speech
2020-12-04T14:07:37 [Debug] SpeechService:StopCurrentSpeech Disposing of active speech
2020-12-04T14:07:37 [Debug] SpeechService:StopCurrentSpeech Stopped current speech
2020-12-04T14:08:08 [Debug] SpeechService:Say Interrupting current speech
2020-12-04T14:08:08 [Debug] SpeechService:speak Selecting voice CereVoice Heather - English (Scotland)
2020-12-04T14:08:08 [Debug] SpeechService:speak {"standardVoice":"CereVoice Heather - English (Scotland)","volume":80,"effectsLevel":35,"distortOnDamage":false,"rate":0,"disablessml":false,"enableicao":false}
2020-12-04T14:08:08 [Debug] SpeechService:speak Feeding normal text to synthesizer: This is how I will sound in your Adder.
2020-12-04T14:08:09 [Debug] SpeechService:Speak Seeking back to the beginning of the stream
2020-12-04T14:08:09 [Debug] SpeechService:Speak Stream length is 107708
2020-12-04T14:08:09 [Debug] SpeechFX:addEffectsToSource Effects level is 35, chorus level is 21, reverb level is 28, echo delay is 50
2020-12-04T14:08:09 [Debug] SpeechFX:addEffectsToSource Extending duration by 500ms
2020-12-04T14:08:09 [Debug] SpeechService:play Starting speech
2020-12-04T14:08:09 [Debug] SpeechService:StartSpeech Checking to see if we can start speech
2020-12-04T14:08:09 [Debug] SpeechService:StartSpeech We can - setting active speech
2020-12-04T14:08:09 [Debug] SpeechService:StartSpeech Playing sound buffer
2020-12-04T14:08:09 [Debug] SpeechService:play Waiting for speech - 00:00:02.9410000
2020-12-04T14:08:12 [Debug] SpeechService:StopCurrentSpeech Stopping active speech
2020-12-04T14:08:12 [Debug] SpeechService:play Finished waiting for speech
2020-12-04T14:08:12 [Debug] SpeechService:StopCurrentSpeech Disposing of active speech
2020-12-04T14:08:12 [Debug] SpeechService:StopCurrentSpeech Stopped current speech
```

No significant delays were noted between `Speak` and `addEffectsToSource` for either voice (same timestamp each case so less than 1s delay, probably significantly less than 1s).

- Perhaps other processes are taking up CPU resources on your local machine? Does this speech delay happen consistently? 
- Does it happen equally with both Cerevoice Caitlin and Cerevoice Heather?

**Comment by Catreina** [comment]
The delay happens consistently with both Heather and Caitlin, and only in Voice Attack / EDDI (both as plugin and not).  It does not occur, for example, with Narrator, or in the TTS settings.

The only thing I have not actually tried to do is create a VM or install on a fresh copy of windows to see if it recurs there. The tests I have done have been normal (with Elite Dangerous running), EDDI only, VA/EDDI only, after reboot, after reboot with safe mode.

**Comment by Catreina** [comment]
Just so you have an update - I did a full uninstall and reinstall of CereVoice TTS engine (again), and now it seems that everything is working.

It -LOOKS- like there was an issue with something around how I installed and purchased the licenses, as earlier today both of the voices I bought disappeared from the TTS menus.  Once I reinstalled everything, it -seems- as though the issue has repaired itself.
